src/Interpreters/InterpreterCreateIndexQuery.cpp

```diff
@@ -39,12 +39,12 @@ BlockIO InterpreterCreateIndexQuery::execute()
 
     }
     // Noop if allow_create_index_without_type = true. throw otherwise
-    if (!create_index.index_decl->as<ASTIndexDeclaration>()->type)
+    if (!create_index.index_decl->as<ASTIndexDeclaration>()->getType())
     {
         if (!current_context->getSettingsRef().allow_create_index_without_type)
         {
             throw Exception(ErrorCodes::INCORRECT_QUERY, "CREATE INDEX without TYPE is forbidden."
-                " SET allow_create_index_without_type=1 to ignore this statements.");
+                " SET allow_create_index_without_type=1 to ignore this statements");
```

Comment:
Why did you remove the period?

Reply:
Generally we do need to put in errors, because some context may be added to the message, and period is added anyway:

```
:) CREATE INDEX idx_tab1_0 on tab1 (col0);

CREATE INDEX idx_tab1_0 ON tab1 col0 GRANULARITY 1

Query id: 526b2cfd-aff6-4ad4-8309-ad312ef5aed2


Elapsed: 0.001 sec. 

Received exception from server (version 24.4.1):
Code: 80. DB::Exception: Received from localhost:9000. DB::Exception: CREATE INDEX without TYPE is forbidden. SET allow_create_index_without_type=1 to ignore this statements.. (INCORRECT_QUERY)
```